web-frontend/src/pages/Home/Home.tsx

```diff
@@ -31,11 +47,20 @@ export default function Home() {
     }
   }, [account, navigate]);
 
-  const unformattedBalance = account?.candidateBalance ?? '0';
-  const balance = parseInt(unformattedBalance);
-  const formattedBalance = formatStandard(balance, Unit.NanoMAS);
-  const address = account?.address ?? '';
-  const formattedAddress = maskAddress(address);
+  // Call handleResize once after component mounts to log formattedAddress on page load
+  useEffect(() => {
+    window.addEventListener('resize', handleResize);
+    handleResize();
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
+
+  useEffect(() => {
+    screenWidth <= breakpoint
+      ? setDisplayedContent(formattedAddress)
+      : setDisplayedContent(address);
+  }, [screenWidth, breakpoint, address, formattedAddress]);
```

Comment:
TBH, I don't think we need all that to deal with this feature.

Reply:
it is a P1

Reply:
I am not saying you don't need to address it. The way you are doing is just not necessary. You can deal with in the ui-kit to be valid for all components and using (I guess, only css).

Why introduce 2 useEffects with listeners!?? 🤔